src/Opaleye/Internal/JSONBuildObjectFields.hs

```diff
@@ -0,0 +1,35 @@
+module Opaleye.Internal.JSONBuildObjectFields
+  ( JSONBuildObjectFields,
+    jsonBuildObjectField,
+    jsonBuildObject,
+    jsonCoalesce
+  )
+where
+
+import Opaleye.Internal.Column (Column (Column))
+import Opaleye.Internal.HaskellDB.PrimQuery (Literal (StringLit), PrimExpr (ConstExpr, FunExpr))
+import Opaleye.Internal.PGTypesExternal (SqlJson)
+
+newtype JSONBuildObjectFields
+  = JSONBuildObjectFields [(String, PrimExpr)]
+
+instance Semigroup JSONBuildObjectFields where
```

Comment:
You'll need to import `Data.Semigroup` explicitly so that it works on GHC 8.0 and 8.2.